### PR TITLE
Initial Firebase Work 

### PR DESCRIPTION
**Pull request by PiersonBro**
[AT-51] and [AT-52]

[AT-51]: https://exam-prep.atlassian.net/browse/AT-51?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AT-52]: https://exam-prep.atlassian.net/browse/AT-52?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by PiersonBro** [comment]
I think this also covers [AT-54]!

File: src/components/questions-box/index.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import styles from "./styles.scss";
+import { Answer, Question, QuestionType } from "../../model/Question";
+import { Exam, getPartialExams } from "../../model/Exam";
+
+const QuestionWriterUI = () => {
+	const [examName, setExamName] = useState("");
+	const [selectedExam, setSelectedExam] = useState<Exam | null>(null);
+	const [question, setQuestion] = useState("");
+	const [answers, setAnswers] = useState(new Map<number, string>());
+	const [correctQuestion, setCorrectQuestion] = useState(0);
```
Comment (tgamlem) [inline src/components/questions-box/index.tsx]: useState(0) will not set a default correct answer. It needs to be a value 1-5

File: src/components/questions-box/index.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+				}
+				placeholder='Answer 5'
+			/>
+			<select
+				value={correctQuestion}
+				onChange={(e) => setCorrectQuestion(parseInt(e.target.value))}
+				name='correctAnswer'
+				id='correctAnswer'
+			>
+				<option value='1'>Question 1</option>
```
Comment (tgamlem) [inline src/components/questions-box/index.tsx:132]: If this dropdown is how we are going to have users select the correct answer, we should change the text from "Question 1" to "Answer 1"

File: src/model/Exam.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+import { set, ref, child, onValue } from "firebase/database";
+import { Answer, Question, QuestionType } from "./Question";
+
+type PartialExam = {
+	lastQuestion: string;
+	qid: number;
+	title: string;
+};
+export function getPartialExams() {
+	const partialExam = ref(database, "/question_set");
```
Comment (tgamlem) [inline src/model/Exam.ts:13]: We should rename from "question_set" to "exam_set" or something similar

File: src/model/Exam.ts [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	questions = Array<Question>();
+	currentQuestion: Question | null = null;
+
+	public writeExam() {
+		if (this.id == null) {
+			this.id = Math.floor(
+				Math.random() * Math.floor(Math.random() * Date.now()),
+			);
+		}
+		const PartialExamRef = ref(database, "/question_set");
```
Comment (tgamlem) [inline src/model/Exam.ts:67]: Same thing here

**Review by tgamlem — APPROVED** [review]
A couple of minor comments, but this is looking good

**Pull request merged**